packages/klighd-core/src/sidebar/sidebar-panel.ts

```diff
@@ -84,4 +101,72 @@ export abstract class SidebarPanel implements ISidebarPanel {
     }
 
     abstract render(): VNode;
+
+    public assignQuickActions():void {
+        this.quickActions = [
+            {
+                key: "center",
+                title: "Center diagram",
+                iconId: "maximize",
+                action: CenterAction.create([], { animate: true }),
+            },
+            {
+                key: "fit",
+                title: "Fit to screen",
+                iconId: "maximize-2",
+                action: this.renderOptionsRegistry.getValue(ResizeToFit) ? undefined : KlighdFitToScreenAction.create(true),
+                state: this.renderOptionsRegistry.getValue(ResizeToFit),
+                effect: () => {
+                        this.actionDispatcher.dispatch(SetRenderOptionAction.create(ResizeToFit.ID, !this.renderOptionsRegistry.getValue(ResizeToFit)));
+                        this.update()
+                }
+            },
+            {
+                key: "layout",
+                title: "Layout diagram",
+                iconId: "layout",
+                action: RefreshLayoutAction.create(),
+            },
+            {
+                key: "refresh",
+                title: "Refresh diagram",
+                iconId: "rotate-cw",
+                action: RefreshDiagramAction.create(),
+            },
+            {
+                key: "export",
+                title: "Export as SVG",
+                iconId: "save",
+                action: RequestExportSvgAction.create(),
+            },
+            {
+                key: "create-bookmark",
+                title: "Bookmark",
+                iconId: "bookmark",
+                action: CreateBookmarkAction.create()
+            },
+            {
+                key: "pin-sidebar",
+                title: this.renderOptionsRegistry.getValueOrDefault(PinSidebarOption) ? "Unpin Sidebar" : "Pin Sidebar",
+                iconId: this.renderOptionsRegistry.getValueOrDefault(PinSidebarOption) ? "lock" : "unlock",
+                action: SetRenderOptionAction.create(PinSidebarOption.ID, !this.renderOptionsRegistry.getValueOrDefault(PinSidebarOption)),
+                state: this.renderOptionsRegistry.getValue(PinSidebarOption)
+            },
+        ];
+        
+
+    }
+    public getQuickAction() :QuickActionOption[]{
+        return this.quickActions;
+    }
+    protected handleQuickActionClick(type: PossibleQuickAction): void {
+        const action = this.getQuickAction().find((a) => a.key === type)?.action;
+
+        if (!action) return;
+
+        this.actionDispatcher.dispatch(action);
+    }
```

Comment:
Methods should be protected here imo.
Methods need documentation like this:
```
/**
 * explain what this method does and what it returns. See other documentation
 */
```

please remember to add one empty line between methods.